kasa/tests/fakeprotocol_smart.py

```diff
@@ -206,7 +206,7 @@ def _set_dynamic_light_effect(self, info, params):
         info["get_dynamic_light_effect_rules"]["enable"] = params["enable"]
         if params["enable"]:
             info["get_device_info"]["dynamic_light_effect_id"] = params["id"]
-            info["get_dynamic_light_effect_rules"]["current_rule_id"] = params["enable"]
+            info["get_dynamic_light_effect_rules"]["current_rule_id"] = params["id"]
```

Comment:
So this was a bug in the test framework? Are we missing a test hence why we missed this?

Reply:
Yes, this was a bug in the tests that got discovered while debugging why the tests were not working :-)